conda/core/solve.py

```diff
@@ -1578,7 +1578,9 @@ def _post_solve_tasks(self,
                 if spec.name in original_prefix_map:
                     final_prefix_map[spec.name] = original_prefix_map[spec.name]
                 # Case B: it was never installed, make sure we don't add it.
-                else:
+                # Note that we special-case python, otherwise there's no path
+                # we can still packages to. This makes `test_create_only_deps_flag` pass.
```

Comment:
```suggestion
                # Python is special-cased here because otherwise the directory
                # tree for other packages is not available for the dependencies.
                # This makes test_create.py::test_create_only_deps_flag pass.
```